test/units/testsuite-71.sh

```diff
@@ -89,10 +89,47 @@ test_chassis() {
     fi
 }
 
+restore_sysfs_dmi() {
+    umount /sys/class/dmi/id
+    rm -rf /run/systemd/system/systemd-hostnamed.service.d
+    systemctl daemon-reload
+    systemctl restart systemd-hostnamed
+}
+
+test_firmware_date() {
+    trap restore_sysfs_dmi RETURN
+
+    # Ignore /sys being mounted as tmpfs
+    mkdir /run/systemd/system/systemd-hostnamed.service.d/
+    cat > /run/systemd/system/systemd-hostnamed.service.d/override.conf <<- EOF
+[Service]
+Environment="SYSTEMD_DEVICE_VERIFY_SYSFS=0"
+EOF
+    systemctl daemon-reload
+
+    # mkdir for s390x
+    mkdir -p /sys/class/dmi/id
```

Comment:
How about to simply skip the test when the system does not have /sys/class/dmi/id ?
```sh
if [[ ! -d /sys/class/dmi/id ]]; then
    echo "/sys/class/dmi/id not found, skipping firmware date tests."
    return 0;
fi
```

Reply:
Makes sense, s390x has no DMI nor does powerpc.